prow/cmd/checkconfig/main.go

```diff
@@ -73,4 +82,121 @@ func main() {
 	if err := pluginAgent.Load(o.pluginConfig); err != nil {
 		logrus.WithError(err).Fatal("Error loading Prow plugin config.")
 	}
+
+	// the following checks are useful in finding user errors but their
+	// presence won't lead to strictly incorrect behavior, so we can
+	// detect them here but don't necessarily want to stop config re-load
+	// in all components on their failure.
+	if err := validateTideRequirements(configAgent, pluginAgent); err != nil {
+		reportWarning(o.strict, err, "Invalid tide merge configuration.")
+	}
+}
+
+func validateTideRequirements(configAgent config.Agent, pluginAgent plugins.PluginAgent) error {
+	lgtmPluginConfig := newOrgRepoConfig(pluginAgent.Config().EnabledReposForPlugin(lgtm.PluginName))
+	approvePluginConfig := newOrgRepoConfig(pluginAgent.Config().EnabledReposForPlugin(approve.PluginName))
+
+	lgtmTideConfig, approveTideConfig := newOrgRepoConfig([]string{}, []string{}), newOrgRepoConfig([]string{}, []string{})
+	for _, query := range configAgent.Config().Tide.Queries {
+		requiresLgtm := sets.NewString(query.Labels...).Has(lgtm.LgtmLabel)
+		requiresApproval := sets.NewString(query.Labels...).Has(approve.ApprovedLabel)
+		for _, org := range query.Orgs {
+			if requiresLgtm {
+				lgtmTideConfig.orgs.Insert(org)
+			}
+			if requiresApproval {
+				approveTideConfig.orgs.Insert(org)
+			}
+		}
+		for _, repo := range query.Repos {
+			if requiresLgtm {
+				lgtmTideConfig.repos.Insert(repo)
+			}
+			if requiresApproval {
+				approveTideConfig.repos.Insert(repo)
+			}
+		}
+	}
+
+	overallTideConfig := orgRepoConfig{
+		orgs:  lgtmTideConfig.orgs.Union(approveTideConfig.orgs),
+		repos: lgtmTideConfig.repos.Union(approveTideConfig.repos),
+	}
+
+	var validationErrs []error
+	validationErrs = append(validationErrs, ensureValidConfiguration(lgtm.LgtmLabel, lgtmTideConfig, overallTideConfig, lgtmPluginConfig))
+	validationErrs = append(validationErrs, ensureValidConfiguration(approve.ApprovedLabel, approveTideConfig, overallTideConfig, approvePluginConfig))
+
+	return errorutil.NewAggregate(validationErrs...)
+}
+
+func newOrgRepoConfig(orgs, repos []string) orgRepoConfig {
+	return orgRepoConfig{
+		orgs:  sets.NewString(orgs...),
+		repos: sets.NewString(repos...),
+	}
+}
+
+type orgRepoConfig struct {
+	orgs, repos sets.String
+}
+
+func (c *orgRepoConfig) items() []string {
+	return c.orgs.Union(c.repos).UnsortedList()
+}
+
+func (c *orgRepoConfig) has(target string) bool {
+	return c.repos.Has(target) || c.orgs.Has(target) || (strings.Contains(target, "/") && c.orgs.Has(strings.Split(target, "/")[0]))
+}
+
+// ensureValidConfiguration enforces rules about tide and plugin config.
+// In this context, a subset is the set of repos or orgs for which a specific
+// plugin is either enabled (for plugins) or required for merge (for tide). The
+// tide superset is every org or repo that has any configuration at all in tide.
```

Comment:
> The tide superset is every org or repo that has any configuration at all in tide.

This seems right, isn't what is being passed to the function. The `overallTideConfig` variable only has the orgs and repos that have either `approve` or `lgtm` specified instead of containing all orgs and repos.